packages/synapse-react-client/src/components/dataaccess/AccessRequirementDashboard.tsx

```diff
@@ -28,97 +30,84 @@ export function AccessRequirementDashboard(
   const { onCreateNewAccessRequirementClicked } = props
 
   const location = useLocation()
+  const urlSearchParams = useMemo(
+    () => new URLSearchParams(location.search),
+    [location.search],
+  )
   const history = useHistory()
 
-  const [nameOrID, setNameOrID] = useState<string>('')
-  const [relatedProjectId, setRelatedProjectId] = useState<string | undefined>(
-    undefined,
+  /**
+   * When an input changes, update the props passed to the table and update the search params.
+   *
+   * Debounced to prevent firing many queries while the user is entering text.
+   */
+  const [nameOrID, setNameOrID] = useDebouncedState<string>(
```

Comment:
Refactor these so the inputs are uncontrolled and the state value is automatically debounced. The effects end up being simpler in the new code.